src/Illuminate/Queue/DatabaseQueue.php

```diff
@@ -288,16 +294,15 @@ protected function markJobAsReserved($job)
      * @param  string  $queue
      * @param  string  $id
      * @return void
+     * @throws \Exception|\Throwable
      */
     public function deleteReserved($queue, $id)
     {
-        $this->database->beginTransaction();
-
-        if ($this->database->table($this->table)->lockForUpdate()->find($id)) {
```

Comment:
I've decided to preserve original code and just fix the transaction management. But I am wondering what is the reason to lock a job for update by `id` when the only action then is delete by `id`.

I cannot see the reason why not just do delete by `id` directly in the transaction.  Any thoughts?

Reply:
- Retry setup: 10 sec
- Worker A picks job 1 and takes 9.9 seconds to run it.
- When worker A is about to delete job 1, worker B wants to pop a job and now job 1 is eligible to be picked.

In this scenario, without the lock I think you run an incredibly small chance of Worker B picking the job right before Worker A deletes it.

Reply:
@deleugpn hmmm. If the retry is 10 second the job is eligible for re-run after 10 seconds worker A claimed it even if the job is still running on worker A. This could be a problem of the timing and probably should not happen. Usually its not desired to process the same job twice and concurrently. I guess one should set retry time so that it is greater than any job that can potentially run. 

If I get it right, this preemption can happen even with locking - before A execution flow reaches `deleteReserved()` it is not locked. The lock protects job from being taken by B only on those few lines. I am not sure this was intended protection for this. If we remove the `find()` completely worker A is more likely to delete it in time. 

On the other hand, imagine the the worker B took the job that is still being processed by A. A will reach `deleteReserved()` and deletes the job even though it is still being processed by A - also not very nice.

Reply:
OK lets wrap this up so we can merge and maybe discuss later for 5.6 patch. Thanks!